### PR TITLE
Fcrepo 2747: Advertising External Content Support

### PR DESCRIPTION
**Pull request by harringj**
**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2747

# What does this Pull Request do?
Adds Accept-External-Content-Handling header to OPTIONS response for binary resources and updates unit tests to ensure header is added only for requests to binary resources.

# What's new?
Adds new header inside FedoraLdp.addLinkAndOptionsHttpHeaders()
Updates unit tests in FedoraLdpTest

# How should this be tested?
In addition to running the FedoraLdpTest suite, you can curl an OPTIONS request to a binary resource and inspect the response headers. "Accept-External-Content-Handling: copy,redirect" should be present among those headers. You can also curl an OPTIONS request to a non-binary resource and verify that the same header does not appear.

E.g., for binary: `curl -i -X POST -H "Content-Type:text/plain" "http://localhost:8080/rest"`
                         `curl -i -X OPTIONS http://localhost:8080/rest/<path/to/binary/resource>`

For non-binary: `curl -X POST "http://localhost:8080/rest/"`
                         `curl -i -X OPTIONS http://localhost:8080/rest/<path/to/resource>`

# Interested parties
@dbernstein 
@bseeger

### CODE REVIEW

**Comment by bseeger** [comment]
I'm wondering what happened with this PR. It looks like it's already merged?  Was there a mistake in the setup?

**Comment by bseeger** [comment]
Joe - was this on a new, clean branch?

**Comment by dbernstein** [comment]
@bseeger :  I think @harringj accidentally created the PR without pushing his code.   Presumably more commits are coming.

**Comment by harringj** [comment]
Yeah, sorry, @bseeger! I had let danny know about this, but should have put a note on the tech channel, too. It was a new, clean branch, but somehow I didn't get my local changes pushed. Working on that now...

**Comment by harringj** [comment]
Ok, @dbernstein and @bseeger, this should finally be ready to go!

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java [hunk 1]
```diff
@@ -830,6 +832,7 @@ private void addLinkAndOptionsHttpHeaders() {
 
         } else if (resource() instanceof FedoraBinary) {
             options = "DELETE,HEAD,GET,PUT,OPTIONS";
+            servletResponse.addHeader(ACCEPT_EXTERNAL_CONTENT, "copy,redirect");
```
Comment (dbernstein) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: This should also support "proxy" as well as copy and redirect.

**Comment by dbernstein** [comment]
Also @harringj : you may need to rebase your changes on master and do a force update on your remote branch.

**Pull request merged**